File: 1636.sort-array-by-increasing-frequency/solution.kt
class Solution {
    fun frequencySort(nums: IntArray): IntArray {
        val count = IntArray(201)
        nums.forEach { count[it + 100]++ }
        return nums.sortedWith(compareBy<Int> { count[it + 100] }.thenByDescending { it }).toIntArray()
    }
}

File: 1636.sort-array-by-increasing-frequency/solution.go
func frequencySort(nums []int) []int {
    h := make(map[int]int)
    for _, n := range nums {
        h[n]++
    }
    slices.SortFunc(nums, func(a, b int) int {
        if d := h[a] - h[b]; d != 0 {
            return d
        }
        return b - a
    })
    return nums
}